firewall_linux.go

```diff
@@ -26,18 +26,19 @@ func arrangeUserFilterRule() {
 	if ctrl == nil || !ctrl.iptablesEnabled() {
 		return
 	}
-	_, err := iptables.NewChain(userChain, iptables.Filter, false)
+	iptable := iptables.GetIptable(iptables.IPv4)
```

Comment:
looks like we will need to add another DOCKER_USER jump rule for ipv6, you can add that as a TODO for now

Reply:
There are also some more of this fixed IPv4 (`iptables.GetIptable(iptables.IPv4)`) location:
resolver_unix.go
service_linux.go
drivers/overlay/encryption.go
drivers/overlay/filter.go

This are mainly sections for overly networks.
It would be really great if this would also work with IPv6 because I think this would make the swarm mode IPv6 NAT compatible.
But there are some section that can not be translated to IPv6 because there are some missing features.
(addrtype LOCAL in ip6tables in combination with route_localnet sys config for IPv6)

I am currently trying to change as much as possible and test if it is working.
If this will not be added in this MR should I add a TODO on any of this `iptables.GetIptable(iptables.IPv4)` calls?

Reply:
sure you can add a TODO above those

Reply:
added TODO